storm-server/src/main/java/org/apache/storm/daemon/nimbus/Nimbus.java

```diff
@@ -1149,6 +1155,18 @@ public void setAuthorizationHandler(IAuthorizer authorizationHandler) {
         this.authorizationHandler = authorizationHandler;
     }
 
+    private CuratorFramework getOrCreateZkClient() {
```

Comment:
I guess Nimbus should have CuratorFramework instance, so don't feel the need to apply lazy initialization with double check locking. Nimbus constructor already has leader elector which relies on ZK.

Reply:
Done.